### PR TITLE
Added missing events

### PR DESCRIPTION
**Pull request by arn-the-long-beard**
Hello :)

First PR here, so happy !

I read CONTRIBUTING.md and it says to look at the todo. My jetbrain likes todo stuff, so I found something to start with.

In this PR I will finish to list the events from there https://developer.mozilla.org/en-US/docs/Web/Events

- [x] ~~Check no events have been missing ( there are so many, `waiting` was missing for example :P  )~~
- [x] ~~Complete the list with respect to the structure shown on the official documentation and previous code~~

++ =>

- [x] Remove unnecessary duplicates.
- [x] Fix whitespaces so it looks nice unicorn (I'm not sure the formatter invoked by cargo make verify is clever enough to fix them automatically inside macros).
- [x] Perhaps remove comments // todo finish this and //WebVR is deprecated, so no need to add it I guess.
- [x] Rebase onto master + force push.
- [x] Remove Draft flag.

Thank you for your inputs !

### CODE REVIEW

**Comment by arn-the-long-beard** [comment]
I just have seen many events have the same names event if they are very different. Also I forgot a "," so I made force push on my branch. Hope it is ok lol.

Anyway, how can we deal with the events with similar names ?

**Comment by arn-the-long-beard** [comment]
By the way, is it normal if the linter give me [unreachable_patterns error](https://doc.rust-lang.org/error-index.html#unreachable_patterns) about the code inside the macro `make_events! `?

**Comment by MartinKavik** [comment]
> By the way, is it normal if the linter give me [unreachable_patterns error](https://doc.rust-lang.org/error-index.html#unreachable_patterns) about the code inside the macro `make_events! `?

No, it's because of events with the same name. We don't distinguish event type in this file - it's basically just a mapper between Rust-friendly names and JS names - it means that we don't need e.g. `ProgressAbort => "abort"` and `AbortController => "abort"` because it's already covered by `Abort => "abort"`. 

However I understand why you tried to write them - look at the new api design: https://github.com/seed-rs/seed/issues/331. (There would be probably the 3. draft without `.msg` or it will be changed in another way in the future before we implement it.)

Also this list of events should be autogenerated in the future, if possible. Typed styles are already autogenerated. You can read [this comment](https://github.com/seed-rs/seed/issues/312) in the issue about attribute name list / new api to learn how it works.

---

So I think the next steps for this PR should be:
1. Remove unnecessary duplicates.
1. Fix whitespaces so it looks nice 🦄 (I'm not sure the formatter invoked by `cargo make verify` is clever enough to fix them automatically inside macros).
1. Perhaps remove comments `// todo finish this` and `//WebVR is deprecated, so no need to add it I guess`.
1. Rebase onto `master` + force push.
1. Remove `Draft` flag.
1. Request review from me (there should be a button on the right side panel or just write a comment / msg in chat)
1. I'll merge when CI passes and everything looks good.

**Comment by arn-the-long-beard** [comment]
Wow ! Thank you so much for your message ! Now things make a little bit more sens to me :+1: 

> cargo make verify

I tried it but it did not work. I ll try again !

Then I ll do 1-4 !

**Comment by MartinKavik** [comment]
3.5) Update CHANGELOG 🙂

**Comment by arn-the-long-beard** [comment]
By the way, should I continue to add events or not ? There are many of theme that still needs to be added.

**Comment by MartinKavik** [comment]
I hope we will be able to generate the list automatically, so don't kill too much time with it - add the ones you need or do you think you'll need in the future. We can create other similar PRs once needed.

File: src/dom_entity_names/events/event_names.rs [hunk 1]
```diff
@@ -72,7 +72,9 @@ make_events! {
     DeviceOrientation => "deviceorientation",
     OrientationChange => "orientationchange",
     SmartcardInsert => "smartcard-insert",
-    SmartCardRemove => "smartcard-remove"
+    SmartCardRemove => "smartcard-remove",
+
+    GoPointerCapture => "gotpointercapture",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:77]: GoPointer.. vs GotPointer.. (typo?)

File: src/dom_entity_names/events/event_names.rs [hunk 2]
```diff
@@ -34,15 +34,47 @@ make_events! {
...
 
-    Change => "change",
+    StateChange => "statechange",
 
-    Input => "input"
+    DeviceMotion => "devicemotion",
+    DeviceOrientation => "deviceorientation",
+    OrientationChange => "orientationchange",
+    SmartcardInsert => "smartcard-insert",
+    SmartCardRemove => "smartcard-remove",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:75]: c vs C

Comment (arn-the-long-beard) [reply src/dom_entity_names/events/event_names.rs:75]: Good catch !

File: src/dom_entity_names/events/event_names.rs (hunk 1 above)
Comment (arn-the-long-beard) [reply src/dom_entity_names/events/event_names.rs:77]: yes :P, I' ll correct it

Comment (arn-the-long-beard) [reply src/dom_entity_names/events/event_names.rs]: I was very tired yesterday I guess :P , I put as resolved :)

**Comment by arn-the-long-beard** [comment]
Hello, by the way, why despite the rebase from Master, github  did still ask me to make a merge from Master ?

File: src/dom_entity_names/events/event_names.rs [hunk 3]
```diff
@@ -34,15 +34,45 @@ make_events! {
...
+    StateChange => "statechange",
+
+    DeviceMotion => "devicemotion",
+    DeviceOrientation => "deviceorientation",
+    OrientationChange => "orientationchange",
+    SmartCardInsert => "smartcard-insert",
+    SmartCardRemove => "smartcard-remove",
 
-    Change => "change",
+    GoPointerCapture => "gopointercapture",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:75]: It looks like the typo wasn't the extra T but the missing one 😄 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/gotpointercapture_event

**Comment by MartinKavik** [comment]
> 
> 
> Hello, by the way, why despite the rebase from Master, github did still ask me to make a merge from Master ?

I don't know - perhaps you've rebased onto your fork's old `master` and not on the Seed repo's `master`, or you forgot to pull new changes to your local Seed's `master` branch, who knows..

However PRs shouldn't contain commits like `Merge branch 'master' into feature/events` because they will be rebased to Seed `master` while I'm "merging" the PR, but I can live with it to not complicate your life 😺

**Comment by arn-the-long-beard** [comment]
> However PRs shouldn't contain commits like Merge branch 'master' into feature/events because they will be rebased to 

Thank you for your input. It does make sens.

`but I can live with it to not complicate your life`

Na na, you can make my life as much complex as you need if it does help me to grow and help the project also :) I will fix it :+1:

File: src/dom_entity_names/events/event_names.rs (hunk 3 above)
Comment (arn-the-long-beard) [reply src/dom_entity_names/events/event_names.rs:75]: I am so sorry Martin. I'll do the correction.

Comment (arn-the-long-beard) [reply src/dom_entity_names/events/event_names.rs]: I removed it because we already have it actually.

**Comment by arn-the-long-beard** [comment]
It seems my tool is caching upstream branch for some reason. I fixed it. Now it is nice and clean.

Thank you for your input again and please continue to push me to improve :)

File: src/dom_entity_names/events/event_names.rs [hunk 4]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+    ReadyStateChange => "readystatechange",ValueChange => "ValueChange",
+
+    Invalid => "invalid",Show => "show",
+
+    SVGAbort => "SVGAbort", SVGError => "SVGError", SVGLoad => "SVGLoad", SVGResize => "SVGResize",SVGScroll => "SVGScroll"  , SVGUnload => "SVGUnload",
+
+    Blocked => "blocked", Success => "success", UpgradeNeeded => "upgradeneeded", VersionChange => "versionchange",
+
+    AfterScriptExecute => "afterscriptexecute", BeforeScriptExecute => "beforescriptexecute",
+    DOMMenuItemActive => "DOMMenuItemActive", DOMMenuteItemInactive => "DOMMEnuItemInactive",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:55]:
```suggestion
    DOMMenuItemActive => "DOMMenuItemActive", DOMMenuItemInactive => "DOMMenuItemInactive",
```

File: src/dom_entity_names/events/event_names.rs [hunk 5]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    PopupHidden=>"popuphidden",PopupHiding => "popuphiding", PopupShowing => "popupshowing", PopupShown => "popupshown",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:57]:
```suggestion
    PopupHidden => "popuphidden", PopupHiding => "popuphiding", PopupShowing => "popupshowing", PopupShown => "popupshown",
```

File: src/dom_entity_names/events/event_names.rs [hunk 6]
```diff
@@ -34,15 +34,43 @@ make_events! {
     DurationChange => "durationchange", Emptied => "emptied", Ended => "ended", LoadedData => "loadeddata",
     LoadedMetaData => "loadedmetadata", Pause => "pause", Play => "play", Playing => "playing", RateChange => "ratechange",
     Seeked => "seeked", Seeking => "seeking", Stalled => "stalled", Suspend => "suspend", TimeUpdate => "timeupdate",
-    VolumeChange => "volumechange",
+    VolumeChange => "volumechange", Waiting => "waiting",
+
+    LoadEnd => "loadend",LoadStart => "loadstart" , Timeout => "timeout",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:39]:
```suggestion
    LoadEnd => "loadend", LoadStart => "loadstart", Timeout => "timeout",
```

File: src/dom_entity_names/events/event_names.rs [hunk 7]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    Change => "change",Storage => "storage",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:41]:
```suggestion
    Change => "change", Storage => "storage",
```

File: src/dom_entity_names/events/event_names.rs [hunk 8]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    Checking => "checking",Downloading=> "downloading", NoUpdate => "noupdate",Obselete => "obsolete", UpdateReady => "updateready",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:43]:
```suggestion
    Checking => "checking", Downloading=> "downloading", NoUpdate => "noupdate", Obselete => "obsolete", UpdateReady => "updateready",
```

File: src/dom_entity_names/events/event_names.rs [hunk 9]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    Broadcast => "broadcast", CheckBoxStateChange => "CheckBoxStateChange", HasChange => "haschange",  Input => "input" , RadioStateChange => "RadioStateChange",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:45]:
```suggestion
    Broadcast => "broadcast", CheckBoxStateChange => "CheckBoxStateChange", HasChange => "haschange",  Input => "input", RadioStateChange => "RadioStateChange",
```

File: src/dom_entity_names/events/event_names.rs [hunk 10]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+    ReadyStateChange => "readystatechange",ValueChange => "ValueChange",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:46]:
```suggestion
    ReadyStateChange => "readystatechange", ValueChange => "ValueChange",
```

File: src/dom_entity_names/events/event_names.rs [hunk 11]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    Invalid => "invalid",Show => "show",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:48]:
```suggestion
    Invalid => "invalid", Show => "show",
```

File: src/dom_entity_names/events/event_names.rs [hunk 12]
```diff
@@ -34,15 +34,43 @@ make_events! {
...
+
+    SVGAbort => "SVGAbort", SVGError => "SVGError", SVGLoad => "SVGLoad", SVGResize => "SVGResize",SVGScroll => "SVGScroll"  , SVGUnload => "SVGUnload",
```
Comment (MartinKavik) [inline src/dom_entity_names/events/event_names.rs:50]:
```suggestion
    SVGAbort => "SVGAbort", SVGError => "SVGError", SVGLoad => "SVGLoad", SVGResize => "SVGResize", SVGScroll => "SVGScroll", SVGUnload => "SVGUnload",
```

**Comment by arn-the-long-beard** [comment]
Well, I did it :) 

Next time I need to do some similar work, I will change my way of doing to be sure that everything is what it should be.

**Comment by MartinKavik** [comment]
I think we can merge it once you squash new commits. And I should suggest all changes in one commit, I think that all repo watchers heard a song of incoming mail notifications 😄

**Comment by arn-the-long-beard** [comment]
I pulled them and then rebased and put them in one commit. Is it good ? :)

**Comment by MartinKavik** [comment]
> 
> 
> I pulled them and then rebased and put them in one commit. Is it good ? :)

Looks good! Merging. I squash commits this way: https://www.youtube.com/watch?v=cr1N8VTRmfM

**Comment by arn-the-long-beard** [comment]
Good! Thank you for the video :)

**Pull request merged**